src/test/java/uk/gov/pay/ledger/transaction/model/CsvTransactionFactoryTest.java

```diff
@@ -116,6 +116,22 @@ public void toMapShouldIncludeExternalMetadataFields() {
         assertThat(csvDataMap.get("key-2 (metadata)"), is("value-2"));
     }
 
+    @Test
+    public void toMapShouldIgnoreDuplicateCaseSensitiveMetadataFields() {
+        TransactionEntity transactionEntity = transactionFixture.withTransactionDetails(
+                new GsonBuilder().create().toJson(ImmutableMap.builder()
+                        .put("external_metadata",
+                                ImmutableMap.builder()
+                                        .put("some-key", "value-1").put("SOME-KEY", "value-2").put("Some-key", "value-1").build())
+                        .build()
+                )).toEntity();
+
+        Map<String, Object> csvDataMap = csvTransactionFactory.toMap(transactionEntity);
+
+        assertThat(csvDataMap.get("some-key (metadata)"), is("value-1"));
+        assertThat(csvDataMap.get("SOME-KEY (metadata)"), is(nullValue()));
```

Comment:
I thought we should only have one header for `some-key` and not have `SOME-KEY`. Looks like this assertion is implying we will have a header with null value. May be change  the assertion to check `some-key` is unique in key set.

Reply:
Agreed 👍 good suggestion will do that